### DISCUSSION TITLE
how to select local file or directory

### DISCUSSION THREAD

**Discussion by wanghaisheng** [emilk/egui#1597] [Q&A] [answered]
let say after click a button,choose specific folder,folder path shows in the ui

**Comment by willser** [comment, marked as answer]
You need `rfd`.Following #987 .

Those code work fine in my project.
````rust
if file_button.clicked() {
    if let Some(path) = rfd::FileDialog::new().pick_file() {
        *path_buf = Some(path);
    }
}
````
If you want to choose a folder,you could use `pick_folder` function.

**Comment by wanghaisheng** [comment]
thanks for you quick answer ,i will try
